pkg/services/object/internal/client/client.go

```diff
@@ -131,9 +137,16 @@ func (x GetObjectRes) Object() *object.Object {
 // Returns:
 //   - error of type *object.SplitInfoError if object raw flag is set and requested object is virtual;
 //   - error of type *apistatus.ObjectAlreadyRemoved if the requested object is marked to be removed.
+//
+// GetObject ignores the provided session if it is not related to the requested object.
 func GetObject(prm GetObjectPrm) (*GetObjectRes, error) {
 	if prm.tokenSession != nil {
-		prm.cliPrm.WithinSession(*prm.tokenSession)
+		// here we ignore session if it is opened for other object since such
+		// request will almost definitely fail. The case can occur, for example,
+		// when session is bound to the parent object and child object is requested.
+		if prm.tokenSession.AssertObject(prm.obj) {
```

Comment:
`if .. != nil && ..AssertObject{ `?

Reply:
Thought that it will be better to have a comment inside some block. Fixed.